mgmt/mcumgr: Drop zephyr_ prefix from functions

de-nordic opened the pull request:
The MCUMgr library is now part of Zephyr, so there is no point
to prefix SMP functions with Zephyr.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

Closes: https://github.com/zephyrproject-rtos/zephyr/issues/49962

~~DNM until dependency https://github.com/zephyrproject-rtos/zephyr/pull/50904 is merged.~~

de-nordic committed:
mgmt/mcumgr: Drop zephyr_ prefix from functions

The MCUMgr library is now part of Zephyr, so there is no point
to prefix SMP functions with Zephyr.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

galak reviewed (dismissed):
This is technically a breaking API change, so should go through the API change process: https://docs.zephyrproject.org/latest/develop/api/api_lifecycle.html

de-nordic commented:
> This is technically a breaking API change, so should go through the API change process: https://docs.zephyrproject.org/latest/develop/api/api_lifecycle.html

OK, can follow that path, although nothing I have changed is available in public documentation, nor here https://docs.zephyrproject.org/latest/develop/api/overview.html.

de-nordic committed:
mgmt/mcumgr: Drop zst_ prefix from smp_transport members

With dropping the zephyr_ prefix, the member prefix zst_ makes
no sense.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

de-nordic committed:
mgmt/mcumgr: Rename zst pointers to smp_transport to smpt

Cosmetic change: the zst was short for zephyr_smp_transport,
now it is just smp_transport so smpt makes more sense.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

de-nordic committed:
tests/mgmt/mcumgr/smp_reassembly: Drop zephyr_ prefix

The mcumgr transport API drops zephyr_ prefixes.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

de-nordic committed:
mgmt/mcumgr: Deprecate zephyr_ prefixed API

The zephyr_ prefixed functions have been marked __deprecated.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

galak commented:
Please update release notes for deprecation.

stephanosio commented:
Please rebase to fix CI failure

de-nordic committed:
doc: releases: Note on deprecation of zephyr_ prefix in SMP API

The commit adds note on deprecation of zephyr_ prefix in SMP API
used by SMP transports, that allow MCUMgr receive and process
SMP frames.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

nordicjm commented on subsys/mgmt/mcumgr/smp.c:
```diff
@@ -116,57 +116,57 @@ zephyr_smp_split_frag(struct net_buf **nb, void *arg, uint16_t mtu)
 	nb = rsp;
 
-	mtu = zst->zst_get_mtu(rsp);
+	mtu = smpt->get_mtu(rsp);
 	if (mtu == 0U) {
 		/* The transport cannot support a transmission right now. */
 		return MGMT_ERR_EUNKNOWN;
 	}
 
```
This code can go once the other PRs are merged

nordicjm commented on include/zephyr/mgmt/mcumgr/smp.h:
```diff
@@ -13,23 +13,29 @@
  */
-typedef int zephyr_smp_transport_out_fn(struct zephyr_smp_transport *zst,
+typedef int smp_transport_out_fn(struct smp_transport *smpt,
 					struct net_buf *nb);
```
nit. alignment

nordicjm commented on subsys/mgmt/mcumgr/smp_uart.c:
```diff
@@ -91,7 +91,7 @@ static int smp_uart_init(const struct device *dev)
 
-	zephyr_smp_transport_init(&smp_uart_transport, smp_uart_tx_pkt,
+	smp_transport_init(&smp_uart_transport, smp_uart_tx_pkt,
 				  smp_uart_get_mtu, NULL, NULL);
```
nit. alignment

carlescufi commented:
We agreed with @de-nordic to follow the [stable API change procedure](https://docs.zephyrproject.org/latest/develop/api/api_lifecycle.html#introducing-incompatible-changes)

nordicjm commented:
Needs rebase

nordicjm commented on doc/releases/release-notes-3.2.rst:
```diff
@@ -98,6 +98,49 @@ Deprecated in this release
+  prefix, deprecating prefixed functions and callback type definitions with the
+  prefix and replacing them with with prefix-less variants.
+  The :c:struct:`zephyr_smp_transport` type, representing transport object,
+  is now replaced with c:struct:`smp_transport`, and the later one is used,
```
```suggestion
  is now replaced with :c:struct:`smp_transport`, and the later one is used,
```

nordicjm commented on doc/releases/release-notes-3.2.rst:
```diff
@@ -98,6 +98,49 @@ Deprecated in this release
+     +-------------------------------------+---------------------------------------+
+     | Deprecated                          | Drop in replacement                   |
+     +=====================================+=======================================+
+     | :C:func:`zephyr_smp_transport_init` | :c:func:`smp_transport_init`          |
```
Not sure if C vs c makes a difference here but rest are lowercase

nordicjm commented on doc/releases/release-notes-3.2.rst:
```diff
@@ -98,6 +98,49 @@ Deprecated in this release
+     +---------------------------------------------+---------------------------------------+
+     | Deprecated                                  | Drop in replacement                   |
+     +=============================================+=======================================+
+     | :C:func:`zephyr_smp_transport_out_fn`       | :c:func:`smp_transport_out_fn`        |
```
As above

nordicjm commented on doc/releases/release-notes-3.2.rst:
```diff
@@ -98,6 +98,49 @@ Deprecated in this release
+     +=============================================+=======================================+
+     | :C:func:`zephyr_smp_transport_out_fn`       | :c:func:`smp_transport_out_fn`        |
+     +---------------------------------------------+---------------------------------------+
+     | :c:func:`zephyr_smp_transport_get_mtu_fn`   | :C:func:`smp_transport_get_mtu_fn`    |
+     +---------------------------------------------+---------------------------------------+
+     | :C:func:`zephyr_smp_transport_ud_copy_fn`   | :c:func:`smp_transport_ud_copy_fn`    |
```
As above

nordicjm commented:
@de-nordic needs rebase

nordicjm commented on subsys/mgmt/mcumgr/smp.c:
```diff
@@ -193,7 +187,7 @@ zephyr_smp_tx_rsp(struct smp_streamer *ns, void *rsp, void *arg)
 static int
-zephyr_smp_process_packet(struct zephyr_smp_transport *zst,
+smp_process_packet(struct smp_transport *smpt,
 			  struct net_buf *nb)
```
nit. alignment, can all go on one line

de-nordic commented:
> @de-nordic please fix the compliance issue

@carlescufi checkpatch wants me to do this
```
-                              zephyr_smp_transport_ud_free_fn *ud_free_func);
+                              zephyr_smp_transport_ud_free_fn * ud_free_func)
```
to correct "problem". Seems weird.

gmarull commented:
> > @de-nordic please fix the compliance issue
> 
> @carlescufi checkpatch wants me to do this
> 
> ```
> -                              zephyr_smp_transport_ud_free_fn *ud_free_func);
> +                              zephyr_smp_transport_ud_free_fn * ud_free_func)
> ```
> 
> to correct "problem". Seems weird.

check `scripts/checkpatch/typedefsfile`, btw why isn't that typedef already a pointer?

nordicjm commented on include/zephyr/mgmt/mcumgr/smp.h:
```diff
@@ -98,17 +131,31 @@ struct zephyr_smp_transport {
  */
-void zephyr_smp_transport_init(struct zephyr_smp_transport *zst,
+void smp_transport_init(struct smp_transport *smpt,
+			       smp_transport_out_fn *output_func,
+			       smp_transport_get_mtu_fn *get_mtu_func,
+			       smp_transport_ud_copy_fn *ud_copy_func,
+			       smp_transport_ud_free_fn *ud_free_func);
```
nit. alignment

nordicjm commented on subsys/mgmt/mcumgr/smp.c:
```diff
@@ -218,54 +211,54 @@ zephyr_smp_process_packet(struct zephyr_smp_transport *zst,
-			  zephyr_smp_transport_ud_copy_fn *ud_copy_func,
-			  zephyr_smp_transport_ud_free_fn *ud_free_func)
+smp_transport_init(struct smp_transport *smpt,
+			  smp_transport_out_fn *output_func,
+			  smp_transport_get_mtu_fn *get_mtu_func,
+			  smp_transport_ud_copy_fn *ud_copy_func,
+			  smp_transport_ud_free_fn *ud_free_func)
```
nit. alignment

de-nordic commented:
> check `scripts/checkpatch/typedefsfile`, btw why isn't that typedef already a pointer?

OK so I have no good reason not to fix the new definitions (the `smp_transport_`), but I do not want to touch the old `zephyr_smp_transport_` definitions which I am deprecating anyway.
@gmarull  Can the compliance issue be waved in this instance, as this has been triggered only because I have extended original declaration of `zephyr_smp_transport_init` to inline it, and actually just call `smp_transport_init`, and have it deprecated.

nordicjm commented:
@de-nordic needs rebase

nordicjm reviewed (changes requested):
smp_dummy.c needs updates too

fabiobaltieri commented:
@de-nordic we are in stabilization phase, should this be considered a bugfix or part of some deprecation we need to do for v3.2? Otherwise, can we wait for v3.3 or you want this to be discussed with the TSC for merging during stabilization?

de-nordic commented:
> @de-nordic we are in stabilization phase, should this be considered a bugfix or part of some deprecation we need to do for v3.2? Otherwise, can we wait for v3.3 or you want this to be discussed with the TSC for merging during stabilization?

@fabiobaltieri  Lets postpone it and have it merged just after the 3.2 gets released; I think it will be more wise to introduce API change at the beginning of development cycle rather than just prior to release.

nordicjm commented:
@de-nordic please rebase
